pyomo/common/deprecation.py

```diff
@@ -247,6 +254,13 @@ def deprecated(msg=None, logger=None, version=None, remove_in=None):
         remove_in (str): the version in which the decorated object will be
             removed from the code.
 
+    Example
+    -------
+    >>> from pyomo.common.deprecation import deprecated
+    >>> @deprecated('pyomo.common.deprecation', version='1.2.3')
+    WARNING: DEPRECATED: pyomo.common.deprecation has been deprecated.
+             (deprecated in 1.2.3)
```

Comment:
`deprecated` is a decorator, so the example needs to decorate something.  Maybe a simple (empty) class or a small function?